src/test/regress/sql/background_task_queue_monitor.sql

```diff
@@ -279,6 +279,32 @@ SELECT job_id, task_id, status FROM pg_dist_background_task
     WHERE task_id IN (:task_id1, :task_id2)
     ORDER BY job_id, task_id;  -- show that task is cancelled
 
+-- TEST11
+-- verify that we do not allow parallel task executors involving a particular node
+-- more than citus.max_parallel_moves_per_node
+ALTER SYSTEM SET citus.max_parallel_moves_per_node = 2;
+SELECT pg_reload_conf();
+BEGIN;
+SELECT nodeid AS worker_1_node FROM pg_dist_node WHERE nodeport=:worker_1_port \gset
+INSERT INTO pg_dist_background_job (job_type, description) VALUES ('test_job', 'simple test to verify max parallel moves per node') RETURNING job_id AS job_id1 \gset
+INSERT INTO pg_dist_background_task (job_id, command, node_tokens) VALUES (:job_id1, $job$ SELECT pg_sleep(5); $job$, ARRAY [:worker_1_node]) RETURNING task_id AS task_id1 \gset
+INSERT INTO pg_dist_background_task (job_id, command, node_tokens) VALUES (:job_id1, $job$ SELECT pg_sleep(5); $job$, ARRAY [:worker_1_node]) RETURNING task_id AS task_id2 \gset
+INSERT INTO pg_dist_background_task (job_id, command, node_tokens) VALUES (:job_id1, $job$ SELECT pg_sleep(5); $job$, ARRAY [:worker_1_node]) RETURNING task_id AS task_id3 \gset
+COMMIT;
+
+SELECT citus_task_wait(:task_id1, desired_status => 'running');
+SELECT citus_task_wait(:task_id2, desired_status => 'running');
+SELECT citus_task_wait(:task_id3, desired_status => 'runnable');
+
+SELECT job_id, task_id, status FROM pg_dist_background_task
+    WHERE task_id IN (:task_id1, :task_id2, :task_id3)
+    ORDER BY job_id, task_id; -- show that last task is not running but ready to run(runnable)
```

Comment:
Needs some more tests. At least the following:
- Tasks with multiple node_tokens
- Increasing max_parallel_moves_per_node during a rebalance
- Decreasing max_parallel_moves_per_node during a rebalance

Reply:
I can't test with a rebalance right now but I am testing with dummy background tasks.